### PR TITLE
Fix flaky tests

### PR DESCRIPTION
**Pull request by shilgam**

### CODE REVIEW

File: tests/features/tags/add-tag.spec.ts [hunk 1]
```diff
@@ -40,89 +40,71 @@ test.describe('Tags', () => {
...
     await test.step(`I mark checkbox 'important'`, async () => {
       await pages.modals.addTag.checkImportant(0);
-      expect(pages.modals.addTag.isCheckboxMarked(0)).toBeTruthy();
     });
     await test.step(`I click 'create' button`, async () => {
       await pages.modals.addTag.addNewTag.click();
       await pages.tags.waitUntilTagVisible(tagName);
       expect(await pages.tags.isTagVisible(`${tagName}`)).toBeTruthy();
-      expect(await pages.tags.isTagImportant(`${tagName}`)).toBeTruthy();
+      await expect(pages.tags.importantTag(tagName)).toBeVisible();
```
Comment (FlySpot) [inline tests/features/tags/add-tag.spec.ts:48]: Are you sure you need 'await' here? The 'importantTag' returns just a Locator, not a Promise.
Also, I recommend using the method .isVisible from CustomElement and changing the assertion on .toBeTruthy

Do it for all places in the MR

File: tests/ui/elements/list.ts [hunk 2]
```diff
@@ -160,9 +123,9 @@ export default class List extends CustomElement {
    * @returns
    */
   async isListItemVisible(identifier: string | number): Promise<boolean> {
-    this.customElement = await this.getListItem(identifier);
-
-    return this.isVisible();
+    const listItem: Locator = await this.getListItem(identifier);
+    await listItem.waitFor();
```
Comment (FlySpot) [inline tests/ui/elements/list.ts:127]: I belive firstly you should wait for the element and then make a get..

**Review by makonakro — DISMISSED** [review]

File: tests/ui/elements/list.ts [hunk 3]
```diff
@@ -217,10 +180,11 @@ export default class List extends CustomElement {
...
 
   /**
-   * Check if at least one item is presented in the list
+   * Check if list is empty
    */
-  async isListNotEmpty(): Promise<boolean> {
-    return this.listItems.isVisible();
+  async isListEmpty(): Promise<boolean> {
+    await this.listItems.first().waitFor({ state: 'hidden' });
+    return this.listItems.isHidden();
```
Comment (FlySpot) [inline tests/ui/elements/list.ts:187]: Looks like overengineering:
.waitFor({ state: 'hidden' }) - if not, we will wait for the global timeout and just after be filed an exception, looks like redundant check in this case

File: tests/ui/pages/management/tags-page.ts [hunk 4]
```diff
@@ -22,6 +23,14 @@ export default class TagsPage extends ManagementPage {
     return new List(this.page, SELECTORS.tagLineRoot, SELECTORS.tagLineItem);
   }
 
+  public normalTag(tagName: string): Locator {
+    return this.page.locator(SELECTORS.tagString(tagName));
+  }
+
+  public importantTag(tagName: string): Locator {
+    return this.normalTag(tagName).locator(SELECTORS.tagImportantMark, { hasText: 'important' });
```
Comment (FlySpot) [inline tests/ui/pages/management/tags-page.ts:31]: you can use PW text selectors like :has-text
https://playwright.dev/docs/selectors#text-selector

File: tests/features/tags/add-tag.spec.ts (hunk 1 above)
Comment (FlySpot) [reply tests/features/tags/add-tag.spec.ts:48]: Yep, you’re right about that two points above. 
My concern is only that the async function is slower than synchronous, which is why I recommend avoiding using it if have another way.

File: tests/ui/elements/list.ts (hunk 3 above)
Comment (FlySpot) [reply tests/ui/elements/list.ts:186]: Yep, you shouldn’t synthetically create a condition when this rule becomes true. From my point of view, this method is redundant cause you can check this from a spec file directly. From your PR
` expect(await pages.catalog.resultsList.isListEmpty()).toBeTruthy();`
I believe can be changed to 
` expect(await pages.catalog.resultsList. listItems.isHidden()).toBeTruthy();`

File: tests/features/tags/add-tag.spec.ts (hunk 1 above)
Comment (shilgam) [reply tests/features/tags/add-tag.spec.ts:48]: > My concern is only that the async function is slower than synchronous

I think I got your point.
I think that using built-in async matchers will not increase execution time.

WHY?

Imagine, there are multiple assertions in the test. One by one:

```
await expect(locator1).toBeVisible();
await expect(locator2).toBeVisible();
await expect(locator3).toBeVisible();
```

In this case, the test will asynchronously wait until the first expectation will be completed (if the `locator1` was not properly loaded to the page).
But it is very likely that other elements (`locator2` and `locator3`) will already be available on the page. And the test will NOT wait a long time as in the case with `locator1`.

Comment (FlySpot) [reply tests/features/tags/add-tag.spec.ts:48]: For this case you can use Promise.all([]) and check all locators.

File: tests/ui/elements/list.ts (hunk 3 above)
Comment (FlySpot) [reply tests/ui/elements/list.ts:186]: Will be fixed in future MR, in accordance with offline discussion.

**Review by FlySpot — APPROVED** [review]

**Review by makonakro — APPROVED** [review]

**Pull request merged**